tools/space-usage/space-usage-schema.json

```diff
@@ -15,25 +15,30 @@
                "type":"string"
             },
             "total_space":{
-               "type":"integer"
+               "type":"integer",
+               "description":"total available space in bytes"
             },
             "used_space":{
-               "type":"integer"
+               "type":"integer",
+               "description":"total used space in bytes"
             },
             "num_files":{
-               "type":"integer"
+               "type":"integer",
+               "description":"total files (optional field)"
```

Comment:
"required": false or true
can be used actually to tell which field is mandatory or not

Reply:
For some reason, just adding this to the item seems to fail (sadly, I'm not a JSON expert) - I did now follow the approach described at http://json-schema.org/example1.html and specified all the items which are required explicitly, which also solves the issue in a programmatic way.

Reply:
Also: Should I squash *all* commits in a single one, or keep the `README` part, the schema part, and the sample part separate?

Reply:
yes, could you squash your commits in a single one ? and also do the pull request against the next branch as well. 

About this, you can have a look on https://github.com/rucio/rucio/pull/333

I extended a bit the contributing notes for the contributors not using the helper script (in favor of hub or the github interface)

Reply:
All done. The other PR against next is #355 , I also adapted the title on github to contain the component, short title, and fix-string.